newIDE/app/src/Utils/GDevelopServices/Project.js

```diff
@@ -0,0 +1,262 @@
+// @flow
+import axios from 'axios';
+import {
+  GDevelopProjectApi,
+  GDevelopProjectResourcesStorage,
+} from './ApiConfigs';
+import { type AuthenticatedUser } from '../../Profile/AuthenticatedUserContext';
+
+export const CLOUD_PROJECT_NAME_MAX_LENGTH = 50;
+
+const projectResourcesClient = axios.create({
+  withCredentials: true,
+});
+
+type CloudProject = {|
+  id: string,
+  name: string,
+  createdAt: string,
+  currentVersion?: string,
+  deletedAt?: string,
+|};
+
+export type CloudProjectWithUserAccessInfo = {|
+  ...CloudProject,
+  lastModifiedAt: string,
+|};
+
+const reAuthenticateAndRetryIfFailed = async <T>(
```

Comment:
nit: maybe be a bit more precise on what "authenticate" means here, it's actually the project credentials.
```suggestion
const refetchCredentialsForProjectAndRetryIfFailed = async <T>(
```